### DISCUSSION TITLE
使用什么构建工具或脚本？

### DISCUSSION THREAD

**Discussion by aqni** [Tr0py/NKU-thesis-template-2020#54] [General]
我认为需要尽可能多地满足下面的条件。
- 跨平台 
- 无额外依赖，最好包含在texlive或默认包含在各个操作系统。
- 上手门槛低，方便后人维护

我能想到的候选
- gunmake/bash 缺点是windows上安装麻烦
- lua 缺点是基本没人会，虽然不难
- python 在windows上比make容易安装，但并不自带

现在我更倾向于使用python 但是它似乎并不经常被用作构建项目。

**Comment by aqni** [comment]
似乎也可以直接使用软连接，我貌似陷入了误区。基于以下两个事实：
- git可以保留软连接
- zip打包软连接时会打包源文件

可以将sty文件或cls文件链接到多处，而不影响git和zip。

因此，可以不提供构建系统。模板zip的打包和示例pdf的编译交由github action完成。

**Reply by Tr0py** [reply]
> 跨平台
无额外依赖，最好包含在texlive或默认包含在各个操作系统。
上手门槛低，方便后人维护

同意

你希望构建系统实现什么目的？你是想要自动编译dtx生成cls和手册吗？还是只是生成zip file和示例pdf？

我觉得最好是直接维护sty/cls，模版打包和示例pdf给github CICD生成。这样windows可以直接用texwork（是这个吗）编辑，linux/macos可以用编辑器编辑tex+bash编译，release出来的zip上传至overleaf可以直接使用。

**Reply by Tr0py** [reply]
如果用github action的话，我可以来写。

**Reply by aqni** [reply]
只要生成示例pdf和将相关文件打包为zip就好了。

我弄个自描述的文件架构出来，就不写构建系统了。

后面github action 就麻烦你了

**Reply by Tr0py** [reply]
ok好的。那你打算保留dtx吗。

“自描述的文件架构”是指什么？

**Reply by aqni** [reply]
不保留dtx。



多个模板分文件夹存储，但是会有公共代码。自描述是指，
- 公共部分单独存放
- 公共部分模板软链接到各模板

从文件系统上看，每个模板是完整的，直接打包这个文件夹就行。

**Reply by aqni** [reply]
不过我还没有测试，xelatex能否正确访问软连接的原始文件。

**Reply by aqni** [reply]
晚上我提个pr先看一下效果
